pkg/securitycontextconstraints/sccadmission/admission_test.go

```diff
@@ -673,33 +674,38 @@ func TestCreateProvidersFromConstraints(t *testing.T) {
 	}
 
 	for k, v := range testCases {
-		// create the admission handler
-		tc := fake.NewSimpleClientset(v.namespace)
-		scc := v.scc()
-
-		// create the providers, this method only needs the namespace
-		attributes := admission.NewAttributesRecord(nil, nil, coreapi.Kind("Pod").WithVersion("version"), v.namespace.Name, "", coreapi.Resource("pods").WithVersion("version"), "", admission.Create, nil, false, nil)
-		_, errs := sccmatching.CreateProvidersFromConstraints(attributes.GetNamespace(), []*securityv1.SecurityContextConstraints{scc}, tc)
-
-		if !reflect.DeepEqual(scc, v.scc()) {
-			diff := diff.ObjectDiff(scc, v.scc())
-			t.Errorf("%s createProvidersFromConstraints mutated constraints. diff:\n%s", k, diff)
-		}
-		if len(v.expectedErr) > 0 && len(errs) != 1 {
-			t.Errorf("%s expected a single error '%s' but received %v", k, v.expectedErr, errs)
-			continue
-		}
-		if len(v.expectedErr) == 0 && len(errs) != 0 {
-			t.Errorf("%s did not expect an error but received %v", k, errs)
-			continue
-		}
+		t.Run(k, func(t *testing.T) {
+			// create the admission handler
+			tc := fake.NewSimpleClientset(v.namespace)
+			scc := v.scc()
+
+			// create the providers, this method only needs the namespace
+			attributes := admission.NewAttributesRecord(nil, nil, coreapi.Kind("Pod").WithVersion("version"), v.namespace.Name, "", coreapi.Resource("pods").WithVersion("version"), "", admission.Create, nil, false, nil)
+			// let timeout based failures fail fast
+			ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+			defer cancel()
+			_, errs := sccmatching.CreateProvidersFromConstraints(ctx, attributes.GetNamespace(), []*securityv1.SecurityContextConstraints{scc}, tc)
+
+			if !reflect.DeepEqual(scc, v.scc()) {
+				diff := diff.ObjectDiff(scc, v.scc())
+				t.Errorf("%s createProvidersFromConstraints mutated constraints. diff:\n%s", k, diff)
+			}
+			if len(v.expectedErr) > 0 && len(errs) != 1 {
+				t.Errorf("%s expected a single error '%s' but received %v", k, v.expectedErr, errs)
+				return
```

Comment:
`t.Fatalf()` here and below with `t.Errorf` followed by a return

Reply:
fixed